### PR TITLE
Inform user about `set_to_none` when using DeepSpeed in Lite

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

The deepspeed optimizer does not accept the argument `optimizer.zero_grad(set_to_none=...)`. This would create an error, but in Fabric, we can automatically drop it and inform the user.

Should we do this? Perhaps it should be left to the user to resolve any deepspeed errors?

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning_fabric/wrappers.py [hunk 1]
```diff
@@ -68,6 +71,10 @@ def step(self, closure: Optional[Callable] = None) -> Any:
             **kwargs,
         )
 
+    def zero_grad(self, **kwargs: Any) -> None:
```
Comment (carmocca) [inline src/lightning_fabric/wrappers.py:74]: I agree with the purpose of the PR, but the implementation is different from what we have always done.

Historically, for this change, we would've rerouted all `zero_grad` calls through the strategy interface so that the `DeepSpeedStrategy` can override this hook and do this parsing.

In this case, you've skipped all that by just doing it directly in our optimizer wrapper.

Maybe this argument is more generic than just deepspeed and it makes sense to have it here. Or maybe this will eventually need to be refactored into a strategy.zero_grad().

I'm okay with this because it's a very simple validation and deepspeed themselves should fix the inconsistency, but this might go away eventually. Up to you if you want to do the full strategy change or keep it like this.

**Review by carmocca — APPROVED** [review]

File: src/lightning_fabric/wrappers.py (hunk 1 above)
Comment (awaelchli) [reply src/lightning_fabric/wrappers.py:74]: Yes exactly, the choice here was based on your last argument. The `zero_grad` is such a simple method right now and there is currently no need for the strategy to abstract anything. If this should change in the future, we will add a public method to the strategy. Until then, Fabric and its wrappers can be that thin "frontend" translation layer.
